### PR TITLE
Fix several .htaccess and .user.ini related problems

### LINKED ISSUE #21146: Mod_php7.c reference missing in .htaccess
Version ownCloud 8.2.1 with PHP7 and httpd/apache.

[owncloud root]/.htaccess does not have reference for mod_php7.c module.

Because of this the following parameters under ifModule are skipped:
> `<IfModule mod_php5.c>`
>  php_value upload_max_filesize 18G
>  php_value post_max_size 18G
>  php_value memory_limit 512M
>  php_value mbstring.func_overload 0
>  php_value always_populate_raw_post_data -1
>  php_value default_charset 'UTF-8'
>  php_value output_buffering 0
>  `<IfModule mod_env.c>`
>    SetEnv htaccessWorking true
>  `</IfModule>`
> `</IfModule>`

This can be verified under admin console where the upload limit value does not change according to php_value upload_max_filesize and php_value post_max_size values in .htaccess.
In addition to this, admin page complains about the lack of privileges and the value field is greyed out.

Replacing mod_php5.c string with mod_php7.c fixes the problem.

### LINKED ISSUE #23144: Code integrity failure when modifying .user.ini
### Steps to reproduce
1. Change core's .user.ini 
2. Reload owncloud

### Expected behaviour

1. I should be able to modify an ini file

2. If `.user.ini` is to be the acting arbiter of `post_max_size` and `upload_max_filesize` on NGINX configurations, then it should not be code signed.. (i.e. You *need* to change it) 

### Actual behaviour

1. Message display at the top of web UI: "There were problems with the code integrity check. More information" 

2. 
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- .user.ini

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [.user.ini] => Array
                        (
                            [expected] => 0a557e3cdca4c2e3675deed761d79d109011dcdebbd9c7f6429f1d3476938ec95729543d7384651d1d0c48e26c5024cc5f517445920915a704ea748bdb903c5f
                            [current] => 2cbd840c4ea40edf706fa45ed23836e7e0757630030280a0f97b76d3ae97429361d05138e4484ef4e53b3c37da3d95b5de19785944ccebcae2d8d35f98091f4b
                        )

                )

        )

)
```
### Server configuration
**Operating system**: CentOS 7

**Web server:** NGINX 1.9.12

**Database:** MariaDB

**PHP version:** PHP 7.0.4 

**ownCloud version:** 9 stable

**Updated from an older ownCloud or fresh install:** 8.2.2

**Where did you install ownCloud from:** repositories

### PR DESCRIPTION
**Pull request by LukasReschke**
This fixes several .htaccess and .user.ini related problems.
## Incompatible with PHP 7

The current `.htaccess` file defines `mod_php5.c` instead of `mod_php7.c`. This leads to the problem that the default values of PHP are used such as 2MB upload etc.

Furthermore in the ownCloud admin interface it was not possible to change the maximum upload size. While there was an edit form it simply had no effect at all since only the value for PHP 5 was edited.

To test this install PHP 7 using mod_php and try to adjust the max upload size in the ownCloud admin menu. Without this patch this was failing.

This is done in https://github.com/owncloud/core/commit/2cfde7cd0b309e4ea959fa828e0c10804cd747b1 and https://github.com/owncloud/core/commit/c7abd307c074c5e4fe1dc99fe4a8b094de9555b9
## Integrity Checker Errors

The `.htaccess` and `.user.ini` can have custom modifications in it which may break the code checker. 
Generally this is not a problem because the files are checked _BEFORE_ they have any edits.

While this is generally not a problem there is one scenario where a integrity checker warning is thrown:
1. Install ownCloud 9.0 with some missing files
2. Integrity checker warning is thrown
3. Edit the `.htaccess` file using the admin menu
4. Fix the problems from step 1. Click rescan. Now the `.user.ini` and `.htaccess` edits from 3 are pointed out. 

So basically as soon as somebody pressed "Rescan" the edited `.htaccess` and `.user.ini` files would be pointed out as error.

This is a little bit harder to test given the fact that one has to sign ownCloud yourself to test this. Anyways, we have new unit tests covering this situation.

This is done in https://github.com/owncloud/core/commit/5278bfe0e49b73a2b9df03df42ca77ef837b16c9

<hr/> 

Fixes https://github.com/owncloud/core/issues/23144 
Fixes https://github.com/owncloud/core/issues/21146

@karlitschek Backport to 9.0.1?

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -37,6 +37,18 @@
     SetEnv htaccessWorking true
   </IfModule>
 </IfModule>
+<IfModule mod_php7.c>
+  php_value upload_max_filesize 513M
+  php_value post_max_size 513M
+  php_value memory_limit 512M
+  php_value mbstring.func_overload 0
+  php_value always_populate_raw_post_data -1
```
Comment (MorrisJobke) [inline .htaccess]: Not needed anymore in PHP7 - see http://php.net/manual/en/ini.core.php#ini.sect.data-handling

Comment (LukasReschke) [inline .htaccess]: Removed with https://github.com/owncloud/core/commit/ee840171929edbeca8eafb32515817406da46051 :)

**Comment by MorrisJobke** [comment]
Tested and works :+1:

File: lib/private/files.php [hunk 2]
```diff
@@ -267,7 +267,7 @@ public static function setUploadLimit($size, $files = []) {
 				$pattern = vsprintf($patternMap['pattern'], [$key]);
 				$setting = vsprintf($patternMap['setting'], [$key, $size]);
 				$hasReplaced = 0;
-				$newContent = preg_replace($pattern, $setting, $content, 1, $hasReplaced);
+				$newContent = preg_replace($pattern, $setting, $content, 2, $hasReplaced);
```
Comment (nickvergessen) [inline lib/private/files.php]: :see_no_evil:

**Comment by nickvergessen** [comment]
:+1:

**Comment by karlitschek** [comment]
backport makes sense. :+1:

**Comment by LukasReschke** [comment]
Backport is at https://github.com/owncloud/core/pull/23270

**Comment by srottschaefer** [comment]
Experiencing the maximum upload size bug myself on Arch. PHP7 module is called php7_module there.
Applied the two changes to .htaccess and file.php with the adjustment of the module name in IfModule but the problem still persists. The values in the .htaccess file get changed but the admin interface resets to PHP default value. If I change the values in php.ini, that's the ones being displayed in the admin interface.
It seems Owncloud/Apache still ignores the settings in .htaccess, any idea why?

**Comment by CHuLoYo** [comment]
Integrity check also fails throwing as invalid files the **php.ini** and **error_log** created by server in some cases. The **error_log** is possible to remove, but **php.ini** it's absolutely necessary in my case. I always have the yellow alert on the top of the owncloud because this files.
